### ISSUE TITLE
Sometimes a lot of garbage is generated in Copilot answers in the accessible view

### ISSUE THREAD

**Issue by lpintes** [microsoft/vscode#213074]
Type: <b>Bug</b>

For example, the query:
@workspace what does method xxx do? generates garbage like this in the answer, copied from the accessible view:
(command:_github.copilot.openSymbolFromReferences?%5B%7B%22%24mid%22%3A1%2C%22path%22%3A%22%2Fhome%2Fpintes%2Fdev%2Feurofly%2Fpkg%2Fdata%2Fsessions.go%22%2C%22scheme%22%3A%22file%22%7D%2C%7B%22line%22%3A78%2C%22character%22%3A14%7D%5D "

VS Code version: Code 1.89.1 (dc96b837cf6bb4af9cd736aa3af08cf8279f7685, 2024-05-07T05:13:33.891Z)
OS version: Windows_NT x64 10.0.19045
Modes:
Remote OS version: Linux x64 6.8.8

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-3720QM CPU @ 2.60GHz (8 x 2594)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|undefined|
|Memory (System)|23.58GB (16.51GB free)|
|Process Argv|--crash-reporter-id 3ea7699c-2979-4d6c-acb2-c8754a7543d6|
|Screen Reader|yes|
|VM|0%|

|Item|Value|
|---|---|
|Remote|SSH: vps.pvagner.sk|
|OS|Linux x64 6.8.8|
|CPUs|AMD EPYC 7H12 64-Core Processor (8 x 3293)|
|Memory (System)|6.00GB (1.10GB free)|
|VM|0%|
</details><details><summary>Extensions (20)</summary>

Extension|Author (truncated)|Version
---|---|---
filter-line|eve|2.0.1
seito-openfile|Fr4|2.2.0
kotlin|fwc|0.2.34
copilot|Git|1.194.0
copilot-chat|Git|0.15.2
go|gol|0.41.4
rest-client|hum|0.25.1
makefile-tools|ms-|0.9.10
vscode-proto3|zxh|0.5.5
remotehub|Git|0.62.0
goto-next-previous-member|mis|0.0.6
jupyter-keymap|ms-|1.1.2
remote-containers|ms-|0.362.0
remote-ssh|ms-|0.110.1
remote-ssh-edit|ms-|0.86.0
remote-wsl|ms-|0.88.2
azure-repos|ms-|0.38.0
remote-explorer|ms-|0.4.3
remote-repositories|ms-|0.40.0
vscode-speech|ms-|0.8.0

(1 theme extensions excluded)

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vspor879:30202332
vspor708:30202333
vspor363:30204092
tftest:31042121
vscod805:30301674
binariesv615:30325510
vsaa593cf:30376535
py29gd2263:31024239
vscaac:30438847
c4g48928:30535728
azure-dev_surveyone:30548225
962ge761:30959799
pythongtdpath:30769146
welcomedialog:30910333
pythonidxpt:30866567
pythonnoceb:30805159
asynctok:30898717
pythontestfixt:30902429
pythonregdiag2:30936856
pythonmypyd1:30879173
pythoncet0:30885854
h48ei257:31000450
pythontbext0:30879054
accentitlementst:30995554
dsvsc016:30899300
dsvsc017:30899301
dsvsc018:30899302
cppperfnew:31000557
dsvsc020:30976470
pythonait:31006305
chatpanelc:31048052
dsvsc021:30996838
724cj586:31013169
pythoncenvpt:31049070
fchga941:31046352
pythonprt:31047983
dwnewjupyter:31046869
26j00206:31048877

```

</details>

**Comment by meganrogge** [comment]
Fixed in insiders, will be released later this week

**Comment by meganrogge** [comment]
related, when I click on a link from within the accessible view, it doesn't open
![Screenshot 2024-05-20 at 12 12 54 PM](https://github.com/microsoft/vscode/assets/29464607/0cc14b72-ccf4-41cc-9716-428a613218fd)

**Comment by meganrogge** [comment]
Same thing happens when I open it in the preview

![Screenshot 2024-05-20 at 12 16 08 PM](https://github.com/microsoft/vscode/assets/29464607/202cfb52-d45d-4dcb-add7-418cc884fa3a)

**Comment by meganrogge** [comment]
I think we should just use a regex to get rid of those. The accessible view should just be rendered as plain text. A user can tab between the links in the chat response, which can be easily accessed by escaping out of the accessible view.

**Comment by meganrogge** [comment]
@jooyoungseo and @rperez030 the paradigm we've set up here is the accessible view is a plaintext representation of a view. If a user wants to navigate the links or other features within it, they can use the preview version, IE the chat response in the chat list. 

Does this make sense to you and do you agree?

In this case, that means I will strip out the command info and "garbage chars". 

For example, here is a before and after the change I'm about to make:

Before:

```md
The terminal is primarily managed in the [`src/vs/platform/terminal`](command:_github.copilot.openRelativePath?%5B%7B%22scheme%22%3A%22file%22%2C%22authority%22%3A%22%22%2C%22path%22%3A%22%2FUsers%2Fmeganrogge%2FRepos%2Fvscode%2Fvscode%2Fsrc%2Fvs%2Fplatform%2Fterminal%22%2C%22query%22%3A%22%22%2C%22fragment%22%3A%22%22%7D%5D "/Users/meganrogge/Repos/vscode/vscode/src/vs/platform/terminal") directory. The main interface for the terminal is defined in [`src/vs/platform/terminal/common/terminal.ts`](src/vs/platform/terminal/common/terminal.ts). 

The terminal is also used in various parts of the codebase. For example, it's used in the debug service as seen in [`src/vs/workbench/api/node/extHostDebugService.ts`](src/vs/workbench/api/node/extHostDebugService.ts), and in the remote URL finder as seen in [`src/vs/workbench/contrib/remote/browser/urlFinder.ts`](src/vs/workbench/contrib/remote/browser/urlFinder.ts). 

The terminal can be displayed in different locations as defined by the [`TerminalLocation`](command:_github.copilot.openSymbolFromReferences?%5B%7B%22%24mid%22%3A1%2C%22fsPath%22%3A%22%2FUsers%2Fmeganrogge%2FRepos%2Fvscode%2Fvscode%2Fsrc%2Fvs%2Fplatform%2Fterminal%2Fcommon%2Fterminal.ts%22%2C%22external%22%3A%22file%3A%2F%2F%2FUsers%2Fmeganrogge%2FRepos%2Fvscode%2Fvscode%2Fsrc%2Fvs%2Fplatform%2Fterminal%2Fcommon%2Fterminal.ts%22%2C%22path%22%3A%22%2FUsers%2Fmeganrogge%2FRepos%2Fvscode%2Fvscode%2Fsrc%2Fvs%2Fplatform%2Fterminal%2Fcommon%2Fterminal.ts%22%2C%22scheme%22%3A%22file%22%7D%2C%7B%22line%22%3A627%2C%22character%22%3A0%7D%5D "src/vs/platform/terminal/common/terminal.ts") enum in [`src/vs/platform/terminal/common/terminal.ts`](src/vs/platform/terminal/common/terminal.ts). It can be displayed in the panel or the editor.

Exit this dialog (Escape).
```

After:
```md
The terminal functionality in your workspace is primarily located in the src/vs/workbench/contrib/terminal directory. Here are some key files:
src/vs/workbench/contrib/terminal/electron-sandbox/terminal.contribution.ts: This file likely contains the code that contributes the terminal functionality to the workbench.
src/vs/workbench/contrib/terminal/electron-sandbox/terminalProfileResolverService.ts: This file appears to be responsible for resolving terminal profiles.
src/vs/workbench/contrib/terminal/electron-sandbox/terminalRemote.ts: This file seems to handle remote terminal functionality.
src/vs/workbench/contrib/terminal/terminal.all.ts: This file likely contains the main terminal functionality.
The terminal is also integrated into other parts of your workspace, such as the debug functionality in src/vs/workbench/contrib/debug/node/terminals.ts and the remote functionality in src/vs/workbench/contrib/remote/browser/urlFinder.ts.

```

**Comment by jooyoungseo** [comment]
Yes, that’s much cleaner. I like this change.

**Comment by meganrogge** [comment]
to verify:

- run a chat response that yields a request with some links and codeblocks
- verify you don't see characters like "7D%2C%7B%22line%22%3A78%2C%22character%22%3A14%7D%5D " but that you still see fenced code blocks

<img width="541" alt="Screenshot 2024-05-30 at 11 27 52 AM" src="https://github.com/microsoft/vscode/assets/29464607/ed8128a8-8d9d-4683-98b5-e8765a204fa6">

**Comment by connor4312** [comment]
This should read fine, but it looks like we're missing a line break after the opening fence:

<img width="901" alt="image" src="https://github.com/microsoft/vscode/assets/2230985/04c52eea-a3f7-4d00-b7cc-fc023e2d90e9">

**Comment by meganrogge** [comment]
Are you doing that in the label or inline chat?

**Comment by accesswatch** [comment]
This is from Chat, meaning:


  1.  Highlight a block of code.
  2.  Press ctrl+I.
  3.  Type “/explain”.
  4.  Wait for Copilot to complete.
  5.  Press Alt+F2 and review the text in the accessible view.



I am thinking that this is not ported to the chat experience yet maybe?



Jeff



From: Megan Rogge ***@***.***>
Sent: Wednesday, June 5, 2024 11:33 PM
To: microsoft/vscode ***@***.***>
Cc: Bishop, Jeff - (jeffbis) ***@***.***>; Comment ***@***.***>
Subject: [EXT] Re: [microsoft/vscode] Sometimes a lot of garbage is generated in Copilot answers in the accessible view (Issue #213074)


External Email

________________________________
Are you doing that in the label or inline chat?

**Comment by lpintes** [comment]
Unfortunately this is back. Should be reopened...
